roles/infra/templates/proxy.j2

```diff
@@ -0,0 +1,13 @@
+{{ ansible_managed | comment }}
+
+server {
+  listen 80 default_server;
+  server_name {{ inventory_hostname }};
+  
+  location / {
+    # Allowing for a `burst` of up to 5 requests beyond the specified rate limit. The `nodelay` parameter ensures that excessive requests beyond the burst limit are immediately rejected with a 503 error response instead of being queued.
+    limit_req zone=limited burst=5 nodelay;
```

Comment:
This seems to me like premature optimisation. I would prefer that we just rely on rate limiting rather than replying with either `429` or `503` for the same protection.

Reply:
The comment is obsolete and leads to a misunderstanding. In any case, it will return a `429` because I have configured it to do so with `limit_req_status 429`, I just forgot to update the comment.

Setting `burst` and `nodelay` in rate limiting is a recommended practice. It provides the following benefits:
- Burst Handling: The burst parameter allows to handle temporary traffic bursts that exceed the rate limit. It provides a buffer to accommodate short-term spikes in user activity or traffic variations without immediate rejection of requests.
- User Experience: By queuing requests within the burst limit, burst improves the user experience during high traffic periods. Instead of rejecting requests beyond the rate limit, it allows them to be processed with a slight delay, ensuring a smoother experience for users.
- Server Stability: Properly configuring burst prevents overwhelming traffic that could affect server stability. By limiting the number of requests that can temporarily exceed the rate limit, it helps maintain a stable server performance even during traffic surges.
- Rate Limit Enforcement: Enabling `nodelay` ensures strict rate limit enforcement. Requests beyond the burst limit are immediately rejected, indicating to clients that they have exceeded the rate limit. This encourages clients to adjust their request frequency, preventing unnecessary queuing and optimizing server resources.

In summary, setting `burst` and `nodelay` strikes a balance between accommodating bursts, enhancing user experience, ensuring server stability, and enforcing rate limits effectively.

Reply:
In our context, I understand that we have a rate limit of 10 requests per second. This `burst` setting enables going up to 15 requests, but on which time period? If this is refreshed per second, I don't understand what is the added value of using `burst` instead of setting a 15 req/s rate limit.
You say that “setting `burst` and `nodelay` in rate limiting is a recommended practice”. Recommended by whom? Do you have any first-hand experience in using these settings?

In general, I am very unhappy with this reply. The first point is purely descriptive, the second point is irrelevant out of context (which “user experience” are we improving when we allow 15 req/s instead of 10 req/s?), the third point is absurd (server stability is provided by rate limiting, not by bursts), and the last point is both descriptive and absurd (I don't see why the client is more encouraged to adjust their request frequency with `burst` than with straight-up rate limiting; if anything, it makes it harder to understand why sometimes they can go up to 12 req/s and sometimes not).

While I understand that using ChatGPT to reply to review comments might save you some energy, it does use significantly more on the reviewer's side. Let's take some time to discuss the usage of this tool. Thanks.

Reply:
We discussed this synchronously and came to several conclusions:

1. Assertive phrasings should be backed and their author (i.e. ChatGPT) disclosed.
2. In the current case, there was an ambiguity in the topic: HTTP status code change or usage of `burst` altogether. This ambiguity led to unnecessarily complex discussions.
3. I had a limited understanding of the way rate limiting in NGINX spreads the limit over time. [This article](https://www.nginx.com/blog/rate-limiting-nginx/#bursts) shared by @ndpnt made it clear that `burst` is indeed relevant for our API use case, as long as we explicitly use `429` in all cases.